Source/Classes/Services/DZNPhotoServiceClient.m

```diff
@@ -158,6 +158,9 @@ - (NSDictionary *)photosParamsWithKeyword:(NSString *)keyword page:(NSInteger)pa
         
         //Default to size medium. Size Large causes some buggy behavior with download times.
         [params setObject:@"'Size:Medium'" forKey:@"ImageFilters"];
+        
+        [params setObject:@(resultPerPage) forKey:@"$top"];
```

Comment:
The 400 was from the presence of the page parameter, which Bing doesn't
accept. Top and skip are like limit and offset in classic programming
terminology. Top x results, having skipped y previous results.

On Wed, May 11, 2016, 11:31 PM Ignacio Romero Zurbuchen <
notifications@github.com> wrote:

> In Source/Classes/Services/DZNPhotoServiceClient.m
> <https://github.com/dzenbot/DZNPhotoPickerController/pull/122#discussion_r62960960>
> :
>
> > @@ -158,6 +158,9 @@ - (NSDictionary *)photosParamsWithKeyword:(NSString *)keyword page:(NSInteger)pa
> >
> >          //Default to size medium. Size Large causes some buggy behavior with download times.
> >          [params setObject:@"'Size:Medium'" forKey:@"ImageFilters"];
> > +
> > +        [params setObject:@(resultPerPage) forKey:@"$top"];
>
> What are $top and $skip for?
> How does this fix the 400 request error?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/dzenbot/DZNPhotoPickerController/pull/122/files/d22daf48c8dff988268137a33c4e4c505f14cd3b#r62960960>
>